pkg/client/rp/relying_party.go

```diff
@@ -249,6 +270,20 @@ func NewRelyingPartyOIDC(ctx context.Context, issuer, clientID, clientSecret, re
 	rp.oauthConfig.Endpoint.AuthStyle = rp.oauthAuthStyle
 	rp.endpoints.Endpoint.AuthStyle = rp.oauthAuthStyle
 
+	if rp.pkce == pkceFromDiscovery {
+		if slices.Contains(
+			discoveryConfiguration.CodeChallengeMethodsSupported,
+			oidc.CodeChallengeMethodPlain,
```

Comment:
This checks for both plain and S256 methods. But as far as I can tell, only S256 is then supported in the discovery handler, should this still accept both or only S256?

cc @@muhlemmer as you suggested to implement this option in #506.